File: nan21.dnet.module.hr/nan21.dnet.module.hr.ui.extjs/src/main/webapp/src/ds/RatingLevelLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.hr.skill.ds.model.RatingLevelLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"clientId", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"active", type:"boolean"},
		{name:"ratingScaleId", type:"int", useNull:true},
		{name:"value", type:"int", useNull:true}
	]
});

Ext.define("net.nan21.dnet.module.hr.skill.ds.model.RatingLevelLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"ratingScaleId", type:"int", useNull:true},
		{name:"ratingScaleId_From",type:"int", useNull:true},
		{name:"ratingScaleId_To",type:"int", useNull:true},
		{name:"value", type:"int", useNull:true},
		{name:"value_From",type:"int", useNull:true},
		{name:"value_To",type:"int", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.hr.skill.ds.param.RatingLevelLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.hr/nan21.dnet.module.hr.presenter/src/main/java/net/nan21/dnet/module/hr/skill/ds/model/RatingLevelLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.hr.skill.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;
import net.nan21.dnet.module.hr.skill.domain.entity.RatingLevel;

@Ds(entity = RatingLevel.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = RatingLevelLovDs.fVALUE) })
public class RatingLevelLovDs extends AbstractTypeLov<RatingLevel> {

    public static final String fRATINGSCALEID = "ratingScaleId";
    public static final String fVALUE = "value";

    @DsField(join = "left", path = "ratingScale.id")
    private Long ratingScaleId;

    @DsField()
    private Integer value;

    public RatingLevelLovDs() {
        super();
    }

    public RatingLevelLovDs(RatingLevel e) {
        super(e);
    }

    public Long getRatingScaleId() {
        return this.ratingScaleId;
    }

    public void setRatingScaleId(Long ratingScaleId) {
        this.ratingScaleId = ratingScaleId;
    }

    public Integer getValue() {
        return this.value;
    }

    public void setValue(Integer value) {
        this.value = value;
    }

}
